sophys_gui/components/list_models.py

```diff
@@ -126,19 +159,30 @@ def argumentsTooltipRender(self, item: dict, argsList: dict):
         (" Delete ", ["item_type"], uidRender, deleteTooltipRender)
     ]
 
-    def __init__(self, re_model, plan_changed, plan_items, row_count, listId, parent=None):
+    def __init__(self, re_model, plan_changed, plan_items, row_count, listId, yml_file_path = None, parent=None):
         super().__init__(parent)
 
         self._re_model = re_model
         self.plan_items = plan_items
         self.row_count = row_count
+        self.yml_file_path = yml_file_path
+        self.config = self.openYaml()
         plan_changed.connect(self.onPlanListChanged)
         self.selected_rows = []
         isHistory = listId == "History"
         if isHistory:
             self.columns = self.columns_history
         else:
             self.columns = self.columns_queue
+    
+    def openYaml(self):
```

Comment:
You repeated this function several times, could you move it to the functions.py file and import it? Passing the self.yml_file_path as a parameter.